kernel/src/engine/arrow_expression/evaluate_expression.rs

```diff
@@ -115,29 +111,63 @@ pub(crate) fn evaluate_expression(
         (Struct(_), _) => Err(Error::generic(
             "Data type is required to evaluate struct expressions",
         )),
-        (Unary(UnaryPredicate { op, expr }), _) => {
+        (Predicate(pred), None | Some(&DataType::BOOLEAN)) => {
+            let result = evaluate_predicate(pred, batch)?;
+            Ok(Arc::new(result))
+        }
+        (Predicate(_), Some(data_type)) => Err(Error::generic(format!(
+            "Unexpected data type: {data_type:?}"
```

Comment:
nit: should we specify we got an unexpected data type for predicate evaluation? give a bit more context?